docs/adr/2024-04-30_008-pparams-update.md

```diff
@@ -0,0 +1,83 @@
+---
+slug: 8
+title: |
+  8. Update of protocol parameters
+authors: [Alexey Kuleshevich]
+tags: [Accepted]
+---
+
+## Status
+
+Accepted
+
+## Context
+
+We need an unified approach for predicting and updating protocol parameters.
+
+Outline of the new approach.
+
+### Shelley through Babbage:
+
+1. Genesis key holders submit votes with proposals for new `PParams`. Depending on the
+   timeing within the epoch those votes will either go into the current or the future
+   proposals bucket.
+2. During the first `4k/f` slots in the the PPUP rule PParamUpdate proposals added to
```

Comment:
This is a bit confusing, I agree. It took me a while to figure it out and hopefully I did understanding it correctly.

Here is the calculation for the stability window:
https://github.com/IntersectMBO/cardano-ledger/blob/63ddaf6f3102e06d26ac2c69b2d8530d83d26a19/eras/shelley/impl/src/Cardano/Ledger/Shelley/StabilityWindow.hs#L10-L22

So, one stability window is `ceiling (3k/f)` number of slots. Two stability windows before the end of the epoch is: `ceiling (3k/f) + ceiling (3k/f)`. Because there is a total of `10k/f` slots in a the epoch the first `n` slots, where `n = 10k/f - 2 * ceiling (3k/f)` slots, which makes `n =< 4k/f` the period that is before the `tooLate` or what I call the point of no return.

Reply:
Thank you, this makes sense!  I  didn't know about the total of `10k/f` slots.